src/components/Modal/BottomDockedModal/index.tsx

```diff
@@ -0,0 +1,227 @@
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import type {ViewStyle} from 'react-native';
+import {BackHandler, DeviceEventEmitter, Dimensions, KeyboardAvoidingView, Modal, View} from 'react-native';
+import {LayoutAnimationConfig} from 'react-native-reanimated';
+import useThemeStyles from '@hooks/useThemeStyles';
+import getPlatform from '@libs/getPlatform';
+import CONST from '@src/CONST';
+import Backdrop from './Backdrop';
+import Container from './Container';
+import type ModalProps from './types';
+
+function BottomDockedModal({
+    testID,
+    animationInDelay,
+    animationInTiming = 300,
+    animationOutTiming = 300,
+    avoidKeyboard = false,
+    coverScreen = true,
+    children,
+    hasBackdrop = true,
+    backdropColor = 'black',
+    backdropOpacity = 0.72,
+    customBackdrop = null,
+    deviceHeight: deviceHeightProp = null,
+    deviceWidth: deviceWidthProp = null,
+    isVisible = false,
+    onModalWillShow = noop,
+    onModalShow = noop,
+    onModalWillHide = noop,
+    onModalHide = noop,
+    onDismiss,
+    onBackdropPress = noop,
+    onBackButtonPress = noop,
+    style,
+    statusBarTranslucent = false,
+    ...props
+}: ModalProps) {
+    const [isVisibleState, setIsVisibleState] = useState(isVisible);
+    const [isContainerOpen, setIsContainerOpen] = useState(false);
+    const [isTransitioning, setIsTransitioning] = useState(false);
+    const [deviceWidth, setDeviceWidth] = useState(() => Dimensions.get('window').width);
+    const [deviceHeight, setDeviceHeight] = useState(() => Dimensions.get('window').height);
+
+    const styles = useThemeStyles();
+
+    const handleDimensionsUpdate = useCallback(() => {
+        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+        if (deviceHeightProp || deviceWidthProp) {
+            return;
+        }
+
+        const deviceWidthTemp = Dimensions.get('window').width;
+        const deviceHeightTemp = Dimensions.get('window').height;
+        if (deviceWidthTemp !== deviceWidth || deviceHeightTemp !== deviceHeight) {
+            setDeviceWidth(deviceWidthTemp);
+            setDeviceHeight(deviceHeightTemp);
+        }
+    }, [deviceWidth, deviceWidthProp, deviceHeight, deviceHeightProp]);
+
+    const onBackButtonPressHandler = useCallback(() => {
+        if (isVisible) {
+            onBackButtonPress();
+            return true;
+        }
+        return false;
+    }, [isVisible, onBackButtonPress]);
+
+    const handleEscape = useCallback(
+        (e: KeyboardEvent) => {
+            if (e.key !== 'Escape' || onBackButtonPressHandler() !== true) {
+                return;
+            }
+            e.stopImmediatePropagation();
+        },
+        [onBackButtonPressHandler],
+    );
+
+    useEffect(() => {
+        const deviceEventListener = DeviceEventEmitter.addListener('didUpdateDimensions', handleDimensionsUpdate);
+        if (getPlatform() === CONST.PLATFORM.WEB) {
```

Comment:
We should have checked this for desktop as well. Not doing so caused https://github.com/Expensify/App/issues/61206